Offline page

tth4vb opened the pull request.

1aurabrown commented on nextjs-app/app/globals.css:
```diff
@@ -13,3 +13,10 @@
 }
+
+@layer utilities {
+  .hover\:cursor-green:hover {
```
what is this meant to do? i think this can be accomplished without adding a new utility class

tth4vb replied:
yup this is redundant, removing it

1aurabrown commented on nextjs-app/app/offline/components/offline-projects.tsx:
```diff
@@ -25,29 +25,36 @@ export type OfflineProjectsProps = {
+  )
 }
 
+// Update the type in the OfflineProject props
```
do we need this comment?

1aurabrown commented on nextjs-app/app/offline/components/offline-projects.tsx:
```diff
@@ -56,29 +63,36 @@ function OfflineProject({ i, project, openIndex, setOpenIndex }: {
     <>
       <motion.header
-	      className="border-t border-black py-2 w-full"
+	      className="border-t border-black py-2 w-full header text-4xl flex justify-between items-center cursor-pointer hover:text-green"
```
is `cursor-pointer hover:text-green` accomplishing the same thing as your new utility class?

1aurabrown commented on nextjs-app/app/offline/components/offline-projects.tsx:
```diff
@@ -90,21 +104,27 @@ function OfflineProject({ i, project, openIndex, setOpenIndex }: {
             return (
-              <li key={image._key} >
-                <Image image={image} />
+              <li key={image._key} className="relative w-full h-[50vh]">
```
we probably want images to show at their intrinsic aspect-ration as much as possible. in the case of two-images side by side this will be harder when they have different aspect ratios, but can be accomplished with flexbox and object-cover

1aurabrown commented on nextjs-app/app/offline/components/offline-projects.tsx:
```diff
@@ -90,21 +104,27 @@ function OfflineProject({ i, project, openIndex, setOpenIndex }: {
-        value={text as PortableTextBlock[]}
-      />}
+      {text?.length && (
+        <div className="prose max-w-none">
```
i never used the `prose` classes before, usually i just use the text sizes set up in the tailwind config and set max-width separately if i want it, but if it does what you want, ok!

1aurabrown commented on nextjs-app/app/offline/page.tsx:
```diff
@@ -50,13 +50,8 @@ export default async function OfflinePage(props: Props) {
-          </h2>
-
-        </div>
+      <div className="h-[200px] md:h-[250px]" />
```
if you need top margin, it can be added to the following element rather than adding an empty spacer div

1aurabrown commented on nextjs-app/tailwind.config.ts:
```diff
@@ -79,6 +79,9 @@ export default {
       minHeight: spacing,
       maxHeight: spacing,
+      cursor: {
+        'green': 'pointer',
```
is `cursor-green` being used anywhere?

tth4vb commented:
@1aurabrown maybe I shouldn't have opened a PR yet, but I wasn't thinking the code was ready for review yet, very much a work in progress - thanks for the comments nonetheless! I'm doing another pass now

1aurabrown commented:
ok, ping me when you’re ready for review!

tth4vb commented:
Ok, I think I've now done a proper pass on this that's ready for review. changes made:
- project formatting within an offline project now has different formats depending on number of images and mobile vs. desktop, and images are their 50% of their native size in sanity to fit better in rows (not sure on that approach tbh, would be curious your thoughts)
- changed logic for collapse and opening interaction per project
- was playing with having a green cursor when hovering over an element that could collapse/open the project row, but scrapped this for now, couldn't get it to work how I envisioned
- addressed your other comments on this PR

1aurabrown commented on nextjs-app/app/_components/Footer.tsx:
```diff
@@ -32,25 +32,22 @@ export function Footer({
+          <div>
+            <ul className="flex items-center md:relative justify-center gap-8">
+              <li className="header text-4xl md:text-6xl text-white md:absolute md:left-0">
+                SA
```
use svg, may have been custom kerning

1aurabrown commented on nextjs-app/app/_components/hero-module.tsx:
```diff
@@ -22,7 +22,7 @@ export default function HeroModule({
       {image && (
         <div 
-          className="relative w-full h-[100vh] mx-auto max-h-[845px] max-w-[954px] leading-[0] text-[0] align-top"
+          className="mt-[100px] relative w-full h-[100vh] mx-auto max-h-[845px] max-w-[954px] leading-[0] text-[0] align-top"
```
Try not to use explicit pixels in class names where avoidable: `mt-25` results in margin top of 100px. Can width be a percentage, or if it is a fixed pixel width, use existing class names like max-w-sm or similar?

tth4vb replied:
the mt example aside, I think this is where maybe my image component is maybe a bit wonky, would love your take on that one

1aurabrown commented on nextjs-app/app/offline/components/offline-projects.tsx:
```diff
@@ -25,47 +25,62 @@ export type OfflineProjectsProps = {
+          className="header text-4xl cursor-pointer hover:text-green"
+          onClick={(e) => {
+            e.stopPropagation();
+            setOpenIndex(isOpen ? null : i);
```
does this mean that if an accordion is open and you click on another one, it just closes the currently open one without opening the other?

tth4vb replied:
no the currently open one does close when you click on an unopened one

1aurabrown commented on nextjs-app/app/offline/page.tsx:
```diff
@@ -6,6 +6,7 @@ import OfflineProjects from "./components/offline-projects"
 import type { Metadata, ResolvingMetadata } from "next";
 
 import { offlineQuery } from "@/sanity/lib/queries";
+import { groq } from "next-sanity";
```
is this being used?

tth4vb replied:
no don't think so, was a relic of another attempt, will remove

1aurabrown commented on nextjs-app/sanity/lib/queries.ts:
```diff
@@ -108,7 +108,16 @@ export const project = `
+      asset,
+      alt
+    }
+  }
```
I will check how this is working in the front-end app

1aurabrown commented on nextjs-app/tailwind.config.ts:
```diff
@@ -79,6 +79,7 @@ export default {
       spacing: spacing,
       minHeight: spacing,
       maxHeight: spacing,
+      
```
unneeded change

tth4vb commented:
@1aurabrown lmk what you think of the small updates I made, and then I will probably merge this one in to focus elsewhere for now - ex. image component, online page, the updates we discussed
